lua/nvim-tree/api.lua

```diff
@@ -222,21 +222,44 @@ Api.fs.copy.absolute_path = wrap_node(wrap_explorer_member("clipboard", "copy_ab
 Api.fs.copy.filename = wrap_node(wrap_explorer_member("clipboard", "copy_filename"))
 Api.fs.copy.basename = wrap_node(wrap_explorer_member("clipboard", "copy_basename"))
 Api.fs.copy.relative_path = wrap_node(wrap_explorer_member("clipboard", "copy_path"))
+---
+---@class NodeEditOpts
+---@field quit_on_open boolean|nil default false
+---@field focus boolean|nil default true
 
 ---@param mode string
 ---@param node Node
-local function edit(mode, node)
+---@param edit_opts NodeEditOpts?
+local function edit(mode, node, edit_opts)
   local file_link = node:as(FileLinkNode)
   local path = file_link and file_link.link_to or node.absolute_path
+  local cur_tabpage = vim.api.nvim_get_current_tabpage()
+
   actions.node.open_file.fn(mode, path)
+
+  edit_opts = edit_opts or {}
+
+  if edit_opts.quit_on_open then
+    view.close(cur_tabpage)
+  end
+
+  local focus = edit_opts.focus == nil or edit_opts.focus == true
+  if not focus then
+    -- if mode == "tabnew" a new tab will be opened and we need to focus back to the previous tab
+    if mode == "tabnew" then
+      vim.cmd(":tabprev")
```

Comment:
```suggestion
      vim.cmd.tabprev()
```

When neovim gets around to exposing these as proper API we will be ready.